tests/provider: list() and map() Configuration Language Function Removals in Terraform 0.15

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

Terraform 0.15 and later no longer includes the `list()` and `map()` configuration language functions. The `tolist()` and `tomap()` configuration language functions are available instead since Terraform 0.12 and later, which means it is safe to adjust our acceptance test configurations to maintain backwards compatibility.

```
=== CONT  TestAccAWSAPIGatewayDeployment_Triggers
resource_aws_api_gateway_deployment_test.go:74: Step 1/4 error: Error running pre-apply refresh:
Error: Error in function call
on config345428254/terraform_plugin_test.tf line 50, in resource "aws_api_gateway_deployment" "test":
50:     redeployment = sha1(join(",", list(
51:       jsonencode(aws_api_gateway_integration.test),
52:     )))
|----------------
| aws_api_gateway_integration.test will be known only after apply
Call to function "list" failed: the "list" function was deprecated in
Terraform v0.12 and is no longer available; use tolist([ ... ]) syntax to
write a literal list.
--- FAIL: TestAccAWSAPIGatewayDeployment_Triggers (1.77s)

=== CONT  TestAccAWSAPIGatewayV2Deployment_Triggers
resource_aws_apigatewayv2_deployment_test.go:80: Step 1/5 error: Error running pre-apply refresh:
Error: Error in function call
on config377298991/terraform_plugin_test.tf line 24, in resource "aws_apigatewayv2_deployment" "test":
24:     redeployment = sha1(join(",", list(
25:       jsonencode(aws_apigatewayv2_integration.test),
26:       jsonencode(aws_apigatewayv2_route.test),
27:     )))
|----------------
| aws_apigatewayv2_integration.test will be known only after apply
| aws_apigatewayv2_route.test will be known only after apply
Call to function "list" failed: the "list" function was deprecated in
Terraform v0.12 and is no longer available; use tolist([ ... ]) syntax to
write a literal list.
--- FAIL: TestAccAWSAPIGatewayV2Deployment_Triggers (2.47s)

=== CONT  TestAccAWSCodeBuildProject_Environment_RegistryCredential
resource_aws_codebuild_project_test.go:2108: Step 1/3 error: Error running pre-apply refresh:
Error: Error in function call
on config820532805/terraform_plugin_test.tf line 97, in resource "aws_secretsmanager_secret_version" "test":
97:   secret_string = jsonencode(map("username", "user", "password", "pass"))
Call to function "map" failed: the "map" function was deprecated in Terraform
v0.12 and is no longer available; use tomap({ ... }) syntax to write a literal
map.
--- FAIL: TestAccAWSCodeBuildProject_Environment_RegistryCredential (6.21s)
```

### Affected Resource(s)

* aws_api_gateway_deployment
* aws_apigatewayv2_deployment
* aws_codebuild_project

### References

* https://github.com/hashicorp/terraform/pull/26818

--- a/aws/resource_aws_api_gateway_deployment_test.go
+++ b/aws/resource_aws_api_gateway_deployment_test.go
@@ -403,9 +403,7 @@ resource "aws_api_gateway_deployment" "test" {
   stage_name        = "tf-acc-test"
 
   triggers = {
-    redeployment = sha1(join(",", list(
-      jsonencode(aws_api_gateway_integration.test),
-    )))
+    redeployment = sha1(jsonencode(aws_api_gateway_integration.test))
   }
 
   lifecycle {

--- a/aws/resource_aws_apigatewayv2_deployment_test.go
+++ b/aws/resource_aws_apigatewayv2_deployment_test.go
@@ -256,10 +256,10 @@ resource "aws_apigatewayv2_deployment" "test" {
   api_id = aws_apigatewayv2_api.test.id
 
   triggers = {
-    redeployment = sha1(join(",", list(
-      jsonencode(aws_apigatewayv2_integration.test),
-      jsonencode(aws_apigatewayv2_route.test),
-    )))
+    redeployment = sha1(jsonencode([
+      aws_apigatewayv2_integration.test,
+      aws_apigatewayv2_route.test,
+    ]))
   }
 
   lifecycle {

--- a/aws/resource_aws_codebuild_project_test.go
+++ b/aws/resource_aws_codebuild_project_test.go
@@ -2723,7 +2723,7 @@ resource "aws_secretsmanager_secret" "test" {
 
 resource "aws_secretsmanager_secret_version" "test" {
   secret_id     = aws_secretsmanager_secret.test.id
-  secret_string = jsonencode(map("username", "user", "password", "pass"))
+  secret_string = jsonencode({ username : "user", password : "pass" })
 }
 `, rName))
 }
@@ -2763,7 +2763,7 @@ resource "aws_secretsmanager_secret" "test" {
 
 resource "aws_secretsmanager_secret_version" "test" {
   secret_id     = aws_secretsmanager_secret.test.id
-  secret_string = jsonencode(map("username", "user", "password", "pass"))
+  secret_string = jsonencode({ username : "user", password : "pass" })
 }
 `, rName))
 }

